### PR TITLE
An attempt at getting the zlib extension module to build

### PR DESCRIPTION
**Pull request by brechtm**
See #5

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,7 @@
 # Current director
 PROJECT_DIR=$(shell pwd)
 
-BUILD_NUMBER=2
+BUILD_NUMBER=3
```
Comment (freakboy3742) [inline Makefile]: For future reference - I'll bump this on in preparation for a new build; you don't have to include the change in your patch.

File: Makefile [hunk 2]
```diff
@@ -100,6 +102,20 @@ downloads/openssl-$(OPENSSL_VERSION).tgz:
 
 
 ###########################################################################
+# zlib
+###########################################################################
+
+# Clean the bzip2 project
```
Comment (freakboy3742) [inline Makefile]: Leftover doc change.

File: Makefile [hunk 3]
```diff
@@ -100,6 +102,20 @@ downloads/openssl-$(OPENSSL_VERSION).tgz:
...
+clean-zlib:
+	rm -rf build/*/zlib-$(ZLIB_VERSION)-* \
+		build/*/zlib
+
+# Download original OpenSSL source code archive.
```
Comment (freakboy3742) [inline Makefile]: Another stale doc reference

File: Makefile [hunk 4]
```diff
@@ -215,6 +232,26 @@ $$(OPENSSL_DIR-$1)/libssl.a $$(OPENSSL_DIR-$1)/libcrypto.a: $$(OPENSSL_DIR-$1)/M
...
+	# Unpack sources
+	mkdir -p $$(ZLIB_DIR-$1)
+	tar zxf downloads/zlib-$(ZLIB_VERSION).tar.gz --strip-components 1 -C $$(ZLIB_DIR-$1)
+	# Configure the build
+	cd $$(ZLIB_DIR-$1) && \
+		CC="$$(CC-$1)" \
+		LDFLAGS="$$(LDFLAGS-$1)" ./configure \
+        --static \
+		--prefix=$(PROJECT_DIR)/build/$2/zlib
+#		--host=$$(MACHINE_SIMPLE-$1)-apple-darwin \     # how to set this for zlib?
```
Comment (freakboy3742) [inline Makefile]: The implication here is that this might not work for other platforms - have you tested this build on iOS?

Comment (brechtm) [inline Makefile]: I have not tested this on iOS.

Comment (brechtm) [inline Makefile]: I tried `make iOS`, which fails with:

    checking for python interpreter for cross build... pyenv: python3.5: command not found

**Comment by freakboy3742** [comment]
Apologies for the long delay in looking at this. I've been held up with other things; when I finally took a look a month ago, Xcode had a bug in it; once *that* problem is fixed, I had to update the patch and build system for the all the macOS 10.13 and iOS 11 changes.

However, I've finally got that code working again, so I've been taking a look at outstanding patches. What was your testing regimen for validating that zlib didn't work? I've done some testing with my build (importing zlib, and running various zlib functions), and zlib appears to work, and doesn't need any additional packages, on either macOS or iOS. Going back, it seems to work on the existing released builds too (at least, it does for me)...

**Comment by freakboy3742** [comment]
Closing on the basis that zlib seems to be working for me without the need for this patch. Please reopen if I've missed something.

**Comment by brechtm** [comment]
Sorry, I can't remember the details and I'm currently taking a break from my open-source project, so I won't be using briefcase in the foreseeable future. I'll let you know if I run into this issue again if I ever get back to it. Cheers!

**Pull request closed without merging**